mbq/client/tests/test_client.py

```diff
@@ -5,19 +5,19 @@
 
 
 class CorrelationIDTestCase(TestCase):
-    def setUp(self):
-        self.client = ServiceClient('https://foo.com/')
 
-    def test_correlation_id_provided(self):
+    def test_get_correlation_id_provided(self):
+        self.client = ServiceClient('https://foo.com/', get_correlation_id=lambda: 'hello-world')
         with patch('requests.get') as requests_mock:
-            self.client.get('/url', correlation_id='hello-world')
+            self.client.get('/url')
             self.assertEquals(1, requests_mock.call_count)
             headers = requests_mock.call_args[1]['headers']
             self.assertEquals('hello-world', headers['X-Correlation-Id'])
 
-    def test_correlation_id_not_provided(self):
+    def test_get_correlation_id_not_provided(self):
+        self.client = ServiceClient('https://foo.com/')
         with patch('requests.get') as requests_mock:
             self.client.get('/url')
             self.assertEquals(1, requests_mock.call_count)
             headers = requests_mock.call_args[1]['headers']
-            self.assertTrue(headers['X-Correlation-Id'])
+            self.assertIsNone(headers.get('X-Correlation-Id'))
```

Comment:
You should add a test with a mutating correlation ID and prove that the correlation ID header changes between requests